python/ambassador/envoy/v2/v2listener.py

```diff
@@ -85,10 +85,10 @@ def type(self) -> Literal['tcp', 'http', 'https']:
         else:
             return 'http'
 
-    def add_host(self, host: IRHost) -> None:
+    def add_host(self, host: IRHost, hostname: Optional[str]=None) -> None:
         if self._log_debug:
-            self._logger.debug(f"      CHAIN UPDATE: add HTTP host: hostname={repr(host.hostname)}")
-        self.hosts[host.hostname] = host
+            self._logger.debug(f"      CHAIN UPDATE: add HTTP host: hostname={repr(hostname or host.hostname)}")
+        self.hosts[hostname or host.hostname] = host
```

Comment:
So we're still going to use the hostname in Host as a fallback?

Reply:
This was changed in the next commit (ee738986c5d8951763bc0b7d1286a53058076974). This `hostname or host.hostname` was a stop-gap because until ee738986c5d8 `V3Chain` didn't have a hostname for TCPMappings, just for Hosts.  But I wanted the `host.hostname→group_host` fix to stand alone as its own commit instead of something subtle being snuck in to a larger refactor. And couldn't flip the two around because without the fix, there wasn't a good way to represent the old behavior in the cleaned up code.